### ISSUE TITLE
[serverless-offline-dynamodb-streams] ECONNREFUSED on SIGINT

### ISSUE THREAD

**Issue by cnuss** [CoorpAcademy/serverless-plugins#197]
Greetings!

I've spent the better part of the day debugging this but I'm afraid I have hit a wall...

# Steps to Reproduce

TL;DR: I've set up a sample repo to demonstrate the issue:
1) `git clone https://github.com/cnuss-test-org-001/microservice-1-sls-rest-api`
2) `cd microservice-1-sls-rest-api`
2) `yarn`
3) `yarn start`
4) once up and running, wait a few seconds, then Ctrl+C to SIGINT

Error from Serverless:

```
...
offline: [HTTP] server ready: http://localhost:3000 🚀
offline:
offline: Enter "rp" to replay the last request
^C
offline: Got SIGINT signal. Offline Halting...
➜  microservice-1-sls-rest-api git:(main) ✗ DynamoDB Local failed to start with code 130
c
 Error ---------------------------------------------------

  Error: read EIO
      at TTY.onStreamRead (internal/stream_base_commons.js:209:20)

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              12.20.1
     Framework Version:         2.57.0 (local)
     Plugin Version:            5.4.5
     SDK Version:               4.3.0
     Components Version:        3.17.1
```

# FYIs
 - You can trigger a stream event locally here: http://localhost:3000/microservice-1/swagger.html (A `POST` to the Example controller will trigger a write and a stream event)

# Observations from my debugging
 - I'm using `dynamodb-local` which is bit of a departure from using https://github.com/mhart/dynalite, however it does support streams
 - It doesn't appear that index.end() gets called on `SIGINT`, I thought this was the issue
 - My theory is that the `DynamodbStreamsReadable` emits one last [https://github.com/CoorpAcademy/serverless-plugins/blob/fb4a8a5bf1f92f38336a46d5bce8e94085aa7d4c/packages/dynamodb-streams-readable/src/index.js#L169](`_read()`) on closing, but I can't understand why, so I have decided to kick it off to you with this issue
